### PR TITLE
feat: Add option to ignore `load_balancer` changes to ECS service

### PR DESCRIPTION
**Pull request by gpdenny**
## Description
Create (another :grimacing:) instance of `aws_ecs_service` to allow users to ignore `load_balancer` changes.

## Motivation and Context
I understand this pattern could be never ending, and result in a module that's very difficult to maintain, but to me it feels like a fairly typical use-case to support Codedeploy with Blue/Green deployment.

I'm not sure if maybe it should be a single variable that controls this, instead of setting both `ignore_task_definition_changes` and `ignore_load_balancer_changes` to true?

## Breaking Changes
No breaking changes, default behaviour is unchanged.

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- Tested all three scenarios where each of the instances of `aws_ecs_service` are used.
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: modules/service/outputs.tf [hunk 1]
```diff
@@ -9,7 +9,7 @@ output "id" {
 
 output "name" {
   description = "Name of the service"
-  value       = try(aws_ecs_service.this[0].name, aws_ecs_service.ignore_task_definition[0].name, null)
+  value       = try(aws_ecs_service.this[0].name, aws_ecs_service.ignore_task_definition[0].name, aws_ecs_service.ignore_task_definition_load_balancer[0].name, null)
```
Comment (antonbabenko) [inline modules/service/outputs.tf:12]: A similar change should be implemented for output `id`.

File: docs/README.md [hunk 2]
```diff
@@ -143,6 +144,62 @@ This could be expanded further to include the entire container definitions argum
   <img src="./images/service.png" alt="ECS Service" width="40%">
 </p>
 
+- When using the above `ignore_task_definition_changes` setting, users can also elect to ignore changes to load balancers by setting `ignore_load_balancer_changes` to `true`. (Note: because of the aforementioned manner in which this psuedo-dynamic ignore change is being managed, changing this value after service creation will cause the entire service to be re-created. Change with caution!) This is intended to support the use of [Blue/Green deployment with CodeDeploy](https://docs.aws.amazon.com/AmazonECS/latest/userguide/deployment-type-bluegreen.html) which changes the the service's load balancer configuration.
+
+```hcl
+
```
Comment (antonbabenko) [inline docs/README.md:150]:
```suggestion
```

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Looks pretty good. Please update `main.tf` in the root module to include the same variables.

**Comment by bryantbiggs** [comment]
I think I am more inclined to add this to the `ignore_task_definition` version of the service rather than create yet another service definition. we can just add the `load_balancer` attribute to the ignore list [here](https://github.com/terraform-aws-modules/terraform-aws-ecs/blob/0f941dfd0dde2c13a2e9af0593ec84af4e3c1ef8/modules/service/main.tf#L391), update the docs, and then we can show in the docs a similar hack like that of updating the service desired count https://github.com/terraform-aws-modules/terraform-aws-ecs/blob/master/docs/README.md#service-1 

So one service is static, all controlled in Terraform, the other one is dynamic for changing values

**Comment by gpdenny** [comment]
Agree @bryantbiggs this sounds less likely for things to get out of hand in future :joy: 
Should we rename the `ignore_task_definition` instance as well if we are adding more to the ignore list? Any preference?

**Comment by bryantbiggs** [comment]
we don't need to rename it at this time since that will be a breaking change. Lets just ensure its properly documented

**Comment by david1437** [comment]
Hi, this feature is very important for my company's use case is there any way I can help to move this forward?

**Comment by gpdenny** [comment]
Hey @david1437 I've been distracted changing day jobs, but I'll still need this in my new role. 
Hoping to tackle some point next week :ok_hand:

**Review by bryantbiggs — APPROVED** [review]

**Comment by bryantbiggs** [comment]
@antonbabenko let me know what you think

File: docs/README.md [hunk 3]
```diff
@@ -143,6 +143,64 @@ This could be expanded further to include the entire container definitions argum
...
+    ignore_task_definition_changes = true
+  }
+
+  resource "aws_lb_target_group" "this" {
+    for_each = {
+      blue = {},
+      green = {}
+    }
+
+    name     = each.key
```
Comment (antonbabenko) [inline docs/README.md:163]:
```suggestion
    name = each.key
```

**Review by antonbabenko — APPROVED** [review]

**Pull request merged**